src/modm/platform/adc/stm32f3/adc_impl.hpp.in

```diff
@@ -196,17 +245,74 @@ modm::platform::Adc{{ id }}::startConversion(void)
 }
 
 bool
-modm::platform::Adc{{ id }}::isConversionFinished(void)
+modm::platform::Adc{{ id }}::isConversionFinished()
 {
 	return static_cast<bool>(getInterruptFlags() & InterruptFlag::EndOfRegularConversion);
 }
 
+void
+modm::platform::Adc{{ id }}::startInjectedConversionSequence()
+{
+	acknowledgeInterruptFlags(InterruptFlag::EndOfInjectedConversion |
+			InterruptFlag::EndOfInjectedSequenceOfConversions);
+
+	ADC{{ id }}->CR |= ADC_CR_JADSTART;
+}
+
+bool
+modm::platform::Adc{{ id }}::setInjectedConversionChannel(uint_fast8_t index, Channel channel,
+														  SampleTime sampleTime)
+{
+	if (index >= 4) {
+		return false;
+	}
+	if (!configureChannel(channel, sampleTime)) {
+		return false;
+	}
+
+	static_assert(ADC_JSQR_JSQ2_Pos == ADC_JSQR_JSQ1_Pos + 6);
+	static_assert(ADC_JSQR_JSQ3_Pos == ADC_JSQR_JSQ2_Pos + 6);
+	static_assert(ADC_JSQR_JSQ4_Pos == ADC_JSQR_JSQ3_Pos + 6);
+
+	const uint32_t pos = ADC_JSQR_JSQ1_Pos + 6 * index;
+	const uint32_t mask = ADC_JSQR_JSQ1_Msk << (6 * index);
+	ADC{{ id }}->JSQR = (ADC{{ id }}->JSQR & ~mask) | (static_cast<uint32_t>(channel) << pos);
+	return true;
+}
```

Comment:
Does `uint_fast8_t` actually generate better assembly than `uint8_t`? I would assume not, since it's passed in a 32-bit register anyways.

Reply:
It doesn't matter. If the parameter is a constant the compiler inlines everything anyway. If not the difference in the assembly is `LDR` vs `LDRB` for the load. `uint8_t` is less characters to type, `uint_fast8_t` expresses the intention that I don't care about the exact number of bits. Both will work.

Reply:
Then use `uint8_t`, it's less confusing.